docs/agent/config.md

```diff
@@ -87,7 +87,191 @@ differently from Agent version 5.
 | `gce_updated_hostname` | v6 behaves like v5 with `gce_updated_hostname` set to true. May affect reported hostname, see [doc][gce-hostname] |
 | `collect_security_groups` | feature still available with the aws integration  |
 
+## [BETA] Encrypted secrets
 
+**This feature is in beta and its options or behaviour might break between
+minor or bugfix releases of the Agent.**
+
+This feature is only available on Linux at the moment.
+
+Starting with the agent 6.3.0 the agent is able to collect encrypted secrets in
+the configurations files from an external source. This allows users to use the
+agent alongside a secret management services (like `Vault` or other).
+
+This section will cover how to set up this feature.
+
+### Encrypting password in check configuration
+
+To fetch a secret from a check configuration simply use the `ENC[]` notation.
+This handle can be used as a *value* of any YAML fields in your configuration
```

Comment:
let's be extremely explicit with what we discussed on the previous PR about the `ENC[]` variable needing to be the _full_ value of the YAML field. An example of something that wouldn't work could help too.